packages/config/src/index.ts

```diff
@@ -275,13 +276,18 @@ export default async (
       : path.join(firstWithWriteAccess([npmGlobalPrefix, os.homedir()]), PNPM_GLOBAL)
     pnpmConfig.dir = path.join(globalDirRoot, LAYOUT_VERSION.toString())
 
+    const npmConfigGlobalBinDir = npmConfig.get('global-bin-dir')
+    if (typeof npmConfigGlobalBinDir === 'string') {
+      fs.mkdirSync(npmConfigGlobalBinDir, { recursive: true })
+    }
+
     pnpmConfig.bin = cliOptions.dir
       ? (
         process.platform === 'win32'
           ? cliOptions.dir
           : path.resolve(cliOptions.dir, 'bin')
       )
-      : globalBinDir(knownGlobalBinDirCandidates, { shouldAllowWrite: opts.globalDirShouldAllowWrite === true })
+      : npmConfigGlobalBinDir || globalBinDir(knownGlobalBinDirCandidates, { shouldAllowWrite: opts.globalDirShouldAllowWrite === true })
```

Comment:
I think if `global-bin-dir` is specified, it should be preferred regardless of system and/or other settings.

Reply:
Do you mean cliOptions.dir should not override `global-bin-dir` in a npmrc?

Reply:
`globalBinDir` won't be called if npmConfig.get('global-bin-dir') is truthy.